src/websockets.rs

```diff
@@ -15,7 +15,7 @@ pub trait WebsocketMessage: std::fmt::Debug {
     fn text(&self) -> Option<&str>;
 
     /// Returns the text (if there is any) of the error
-    fn error_message(&self) -> Option<&str>;
+    fn error_message(&self) -> Option<String>;
```

Comment:
Note from the other PR that still isn't resolved here:

> Ah, is this the allocation you were talking about on discord? I thought you were talking about the text function.
>
> Wonder if it would be worth returning an Option<Cow<'a, str>> so that we don't force the libs that don't need to allocate to do so.

Reply:
Had a go at making it a `Cow<str>`, not sure if I did it right